### ISSUE TITLE
The inventory keyword in /etc/ansible/ansible.cfg is not backwards compatible anymore

### ISSUE THREAD

**Issue by jean-christophe-manciot** [ansible/ansible#25268]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

ansible.cfg
##### ANSIBLE VERSION

```
ansible --version
ansible-playbook 2.4.0 (devel 45a9f96774) last updated 2017/06/01 18:48:42 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/actionmystique/Ansible/git-yang-networkop/ansible-101/library']
  ansible python module location = /home/actionmystique/src/Ansible/git-ansible/lib/ansible
  executable location = /home/actionmystique/src/Ansible/git-ansible/bin/ansible-playbook
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```
##### CONFIGURATION

inventory = ./hosts
library = /home/actionmystique/Ansible/git-yang-networkop/ansible-101/library
forks = 1000
gathering = explicit
gather_timeout = 10
roles_path = /home/actionmystique/Ansible/Roles/roles
private_role_vars = yes
hash_behaviour = merge
log_path = /var/log/ansible.log
retry_files_enabled = False
show_custom_stats = True
timeout = 10
pipelining = True
connect_timeout = 30
connect_retries = 30
connect_interval = 1

##### OS / ENVIRONMENT
- **Local host**: Ubuntu 17.04 4.10

##### SUMMARY

The inventory keyword in /etc/ansible/ansible.cfg is not read anymore.
##### STEPS TO REPRODUCE
**Configuration: ansible.cfg**
```
...
inventory      = ./hosts
...
```

**Playbook**: cisco_init_ssh.yml

```
- name: Configuring SSH & SCP for the Cisco device
  hosts:
    - ios_xrv
    - asav
  roles:
    - cisco_init_ssh
```

**Local hosts** file:
```
Cisco-IOS-XR/6.1.2/Ansible# ll hosts
-rwxrwx--- 1 root libvirtd 3758 May 17 15:07 hosts*
Cisco-IOS-XR/6.1.2/Ansible# cat hosts
...
[spines_sf]
XRv_Spine_01 ansible_host=172.21.100.201 os_family=ios_xrv
...
```

##### EXPECTED RESULTS

- Ansible should find the expected hosts file in the current folder, instead of looking into /etc/ansible.

##### ACTUAL RESULTS

```
Using /etc/ansible/ansible.cfg as config file
Parsed /etc/ansible/hosts inventory source with ini plugin
 [WARNING]: provided hosts list is empty, only localhost is available
 [WARNING]: Could not match supplied host pattern, ignoring: XRv_Spine_01
```

**Comment by jean-christophe-manciot** [comment]
**No** such issue with ansible 2.3.1.0-0.1.rc1.

**Comment by bcoca** [comment]
> Parsed /etc/ansible/hosts inventory source with ini plugin

^ it seems to clearly read the expected file, path is relative to config, not cwd, if you pass `-i ./hosts`, then the path is relative to cwd

**Comment by jean-christophe-manciot** [comment]
Then that's what has been changed: **the path used to be relative to cwd** (which makes sense), not /etc/ansible, otherwise how can we tell ansible to read the hosts file in the cwd with a global setting, instead of having to pass the info at each call with -i?
I have a hosts file per project.

**Comment by resmo** [comment]
I am also able to reproduce with a ansible.cfg in cwd which has `inventory = ./hosts`, breaks backwards compatibility.

**Comment by JayKayy** [comment]
@jean-christophe-manciot You could also place an ansible.cfg in each directory from which you're running ansible commands. This would allow you to specify the absolute path for `inventory` on a per directory basis. If you're changing around directories and intending different configuration behaviour for each that sounds like it may be useful to you.

Options for configuration files:

Global(default)
`/etc/ansible/ansible.cfg`

Home dir
`~/.ansible.cfg`

Directory where ansible commands are being run
`./ansible.cfg`

Best,

John

**Comment by jean-christophe-manciot** [comment]
Your solution means replicating the same information in all the folders where we use Ansible.
The truth is, ansible's developers has changed the inventory keyword signification, making it not backwards compatible..

**Comment by resmo** [comment]
@jean-christophe-manciot keep calm. it is a bug. Taking care.

**Comment by jean-christophe-manciot** [comment]
I feel calm & I'm glad to hear that it is taken care of.

**Comment by kumaran85** [comment]
I am using ansible on 2.3.1.0. added my hosts in /etc/ansible/hosts. if i do, --list-hosts below error coming
ERROR! the field 'hosts' is required but was not set. In ansible.cfg inventory path is same. Can anyone help on this?

**Comment by kumaran85** [comment]
I am using ansible on 2.3.1.0. added my hosts in /etc/ansible/hosts. if i do, --list-hosts below error coming
ERROR! the field 'hosts' is required but was not set. In ansible.cfg inventory path is same. Can anyone help on this?

**Comment by kumaran85** [comment]
And using on AWS

**Comment by bcoca** [comment]
@kumaran85 that is totally unrelated to this issue, you are missing `hosts:` in your play, please post to IRC/mailing list instead of on random issues.

**Comment by bcoca** [comment]
this issues was 'fixed' by #12797 which introduces `{{CWD}}` token for those that are looking for that specific behaviour.

**Comment by kumaran85** [comment]
Thanks for your update......I have bit confusion on this. Could you tell me which file needs to be modified?
